consensus/istanbul/core/request.go

```diff
@@ -42,7 +42,8 @@ func (c *core) handleRequest(request *istanbul.Request) error {
 	}
 
 	// Must go through startNewRound to send proposals for round > 0 to ensure a round change certificate is generated.
-	if c.current.State() == StateAcceptRequest && c.current.Round().Cmp(common.Big0) == 0 {
+	// Also ignore requests when not validating.
+	if c.backend.IsValidating() && c.current.State() == StateAcceptRequest && c.current.Round().Cmp(common.Big0) == 0 {
```

Comment:
can we make it that if it's a replica "request" won't even get here? They are redirected somewhere else, in a higher layer (the one that handles switching from replica core impl or primary core impl)

Reply:
This is a leftover. Because the core is properly stopped, the event handler should be stopped which in turn won't pass requests to `handleRequest`